src/wallet/wallet.ts

```diff
@@ -749,15 +927,37 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
 
     const dataToSignHash = tx.getDataToSignHash();
 
+    if (!newOptions.pinCode) {
+      throw new Error('PIN not specified in prepareCreateNewToken options');
+    }
+
+    const xprivkey = wallet.getXprivKey(newOptions.pinCode);
+
     for (const [idx, inputObj] of tx.inputs.entries()) {
-      const inputData = this.getInputData(dataToSignHash, utxosAddressPath[idx]);
+      const inputData = this.getInputData(
+        xprivkey,
+        dataToSignHash,
+        HathorWalletServiceWallet.getAddressIndexFromFullPath(utxosAddressPath[idx]),
+      );
       inputObj.setData(inputData);
     }
 
     tx.prepareToSend();
     return tx;
   }
 
+  /**
+   * Expects a BIP44 path at the address level and returns the address index
+   *
+   * @param {string} fullPath - The full BIP44 path for the address index
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  static getAddressIndexFromFullPath(fullPath: string): number {
+    return parseInt(fullPath.split('/')[5], 10);
```

Comment:
I'd add the following assert here:

```js
const parts = fullPath.split('/');
assert.equal(6, parts.length);
```

Reply:
Done in 2f9f675beb39a7f6884a5370a94cce05721dfae1

Reply:
Great! Thanks!